compiler/rustc_hir_analysis/src/check/mod.rs

```diff
@@ -546,3 +554,80 @@ fn bad_non_zero_sized_fields<'tcx>(
 pub fn potentially_plural_count(count: usize, word: &str) -> String {
     format!("{} {}{}", count, word, pluralize!(count))
 }
+
+pub fn check_function_signature<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    cause_code: ObligationCauseCode<'tcx>,
+    fn_id: LocalDefId,
```

Comment:
This can be a `DefId`, and you can use `CRATE_DEF_ID` for the cases that it's required to be a `LocalDefId`.